apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -13,87 +14,221 @@
          terminate/2,
          code_change/3]).
 
+%% Fetching tx history
+-export([get_txs_since/2]).
+
 -define(GEN_SERVER_OPTS, []).
 
 -record(st, { parent
             , chan_id
+            , chan_vsn
+            , last_block                  %% last time we updated channel vsn
+            , last_top                    %% the block hash of the last check
+            , tx_log = aesc_window:new()
+            , rpt_log = aesc_window:new()
             , closing = false
-            , requests = [] }).
+            , requests = []
+            , opts = #{} }).
+
+-type mode()       :: close
+                    | unlock
+                    | tx_hash
+                    | watch.
+
+-type req_info() :: #{ type         := any()
+                     , parent       := pid()
+                     , callback_mod := module() }.
+
+-type close_req() :: #{ mode      := close
+                      , min_depth := aec_blocks:height()
+                      , info      := req_info() }.
+-type unlock_req() :: #{ mode := unlock
+                       , info := req_info() }.
+-type watch_req() :: #{ mode := watch
+                      , info := req_info() }.
+-type tx_req() :: #{ mode      := tx_hash
+                   , tx_hash   := tx_hash()
+                   , min_depth := aec_blocks:height()
+                   , info      := req_info() }.
 
+-type req() :: close_req() | unlock_req() | watch_req() | tx_req().
+
+-type block_hash() :: binary().
+-type tx_hash()    :: binary().
+-type scenario()   :: top | next_block | fork_switch | {has_tx, tx_hash()}.
+-type chan_vsn()   :: undefined | { aesc_channels:round()
+                                  , aesc_channels:solo_round()
+                                  , aesc_channels:is_active()
+                                  , aesc_channels:locked_until()
+                                  , aesc_channels:state_hash() }.
+-type ch_status() :: undefined | #{ is_active    := aesc_channels:is_active()
+                                  , vsn          := chan_vsn()
+                                  , changed      := boolean()
+                                  , channel      := aesc_channels:channel()
+                                  , lock_period  := aesc_channels:lock_period()
+                                  , locked_until := aesc_channels:locked_until() }.
+
+%% (Commented out since dialyzer complains, but can't really explain why)
```

Comment:
If you paste the error I am happy to help.

I find maps more difficult to follow than records, for the lack of machine-parseable structure. Having dialyzer to perform the basic sanity check on the map would help.

Reply:
Uncommenting dialyzer passed for me.

Reply:
See #2478

Reply:
Addressed in 6ca8c386d (was https://github.com/aeternity/aeternity/pull/2478/commits/5342ead4f720c5854461a1d7fb875e4252cdc189)